core/src/main/java/org/sonarsource/sonarlint/core/container/storage/StorageContainer.java

```diff
@@ -144,9 +146,17 @@ public ModuleUpdateStatus getModuleUpdateStatus(String moduleKey) {
     return getComponentByType(IssueStoreReader.class).getServerIssues(moduleKey, filePath);
   }
 
-  public Iterator<ServerIssue> downloadServerIssues(String moduleKey, String filePath) {
-    // TODO
-    return getServerIssues(moduleKey, filePath);
+  public Iterator<ServerIssue> downloadServerIssues(ServerConfiguration serverConfig, String moduleKey, String filePath) {
+    IssueStoreReader issueStoreReader = getComponentByType(IssueStoreReader.class);
+    StorageManager storageManager = getComponentByType(StorageManager.class);
+    IncrementalUpdate updater = IncrementalUpdate.create(storageManager, serverConfig, issueStoreReader);
+
+    try {
+      updater.update(moduleKey, filePath);
+    } catch (Exception e) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Either log or rethrow this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1166)